packages/lib/server-only/template/create-document-from-template.ts

```
@@ -266,19 +272,23 @@ export const createDocumentFromTemplate = async ({
       }
 
       fieldsToCreate = fieldsToCreate.concat(
-        fields.map((field) => ({
-          documentId: document.id,
-          recipientId: recipient.id,
-          type: field.type,
-          page: field.page,
-          positionX: field.positionX,
-          positionY: field.positionY,
-          width: field.width,
-          height: field.height,
-          customText: '',
-          inserted: false,
-          fieldMeta: field.fieldMeta,
-        })),
+        fields.map((field) => {
+          const prefillValue = prefillValues?.find((value) => value?.id === field.id);
```

if the field doesn't exist, the app doesn't throw an error. it simply proceeds further without adding/modifying any field.

---

should we throw an error if the user tries to pre-fill a field that doesn't exist?

---

Should throw an error to let the user know that it didn't work as intended

---

Also, validate that the field type matches the field meta data